### PR TITLE
Classifier: Move MetaTools into same grid area as subject viewer

### PR DESCRIPTION
**Pull request by goplayoutside3**
## Package
lib-classifier

## Linked Issue and/or Talk Post
Includes #4019 
~~Includes #4005~~

## Describe your changes

This PR moves the MetaTools component into the same grid area as the SubjectViewer, which allows its buttons to always display directly below the subject image regardless of viewport aspect ratio. Screenshots below.

This PR also includes refactoring from class to functional components and RTL unit tests for Metadata, MetadataModal, and MetadataButton.

Previously, the MetaTools buttons displayed below the toolbar height:
old

The changes in this PR display them like this:
new

## How to Review

Run `yarn bootstrap` with this branch and view any project with a subject image. Examples:
    - [i-fancy-cats](https://local.zooniverse.org:3000/projects/brooke/i-fancy-cats/classify/workflow/693)
    - [TESS](https://local.zooniverse.org:3000/projects/nora-dot-eisner/planet-hunters-tess/classify/workflow/11235?env=production)
    - [Poets & Lovers](https://local.zooniverse.org:3000/projects/pmlogan/poets-and-lovers/classify/workflow/21362/subject-set/104809?env=production)

I also added a story for the MetaTools container with a specific case for smaller subject viewer styling.

See corresponding ImageToolbar refactoring in #4008.

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General
- [x] Tests are passing locally and on Github
- [x] Documentation is up to date and changelog has been updated if appropriate
- [x] You can `yarn panic && yarn bootstrap` or `docker-compose up --build` and FEM works as expected
- [x] FEM works in all major desktop browsers: Firefox, Chrome, Edge, Safari (Use Browserstack account as needed)
- [x] FEM works in a mobile browser

## General UX
Example Staging Project: [i-fancy-cats](https://local.zooniverse.org:3000/projects/brooke/i-fancy-cats)
- [x] All pages of a FEM project load: Home Page, Classify Page, and About Pages
- [x] Can submit a classification
- [x] Can sign-in and sign-out
- [x] The component is accessible
  - Can be used with a screen reader [BBC guide to VoiceOver](https://bbc.github.io/accessibility-news-and-you/assistive-technology/testing-steps/voiceover-mac.html)
  - Can be used from the keyboard [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - It is passing accessibility checks in the storybook

## Refactoring
- [x] The PR creator has described the reason for refactoring
- [x] The refactored component(s) continue to work as expected

### CODE REVIEW

**Comment by eatyourgreens** [comment]
This is going to break `MetaTools` on desktop, where it is supposed to extend under the Field Guide button. 

Maybe small screens need to define a new grid layout, not use `DefaultLayout`.

**Comment by eatyourgreens** [comment]
New layouts can be added to the classifier here:
https://github.com/zooniverse/front-end-monorepo/blob/d592efdfd5bfdb2fc7cae92539446d5b03f3c7a5/packages/lib-classifier/src/components/Classifier/components/Layout/helpers/getLayout/getLayout.js#L3-L5

See also Shaun’s ADR about layouts. 
https://github.com/zooniverse/front-end-monorepo/blob/master/docs/arch/adr-32.md

**Comment by goplayoutside3** [comment]
1. Noting that this change was initially requested by @seanmiller26 during development of custom video controls https://github.com/zooniverse/front-end-monorepo/pull/3684#pullrequestreview-1117428037. Further discussion is planned around layouts, so no rush to merge this PR while waiting for a final design decision.


2. > This is going to break `MetaTools` on desktop, where it is supposed to extend under the Field Guide button.

    On desktop, when I view SuperWASP Black Hole hunters on [production](https://www.zooniverse.org/projects/hughdickinson/superwasp-black-hole-hunters/classify/workflow/17386) and [locally](https://local.zooniverse.org:3000/projects/hughdickinson/superwasp-black-hole-hunters/classify/workflow/17386?env=production), the classify page looks the same. When you say "break", do you mean the styling of MetaTools will respond differently to varying viewport widths?

**Comment by eatyourgreens** [comment]
Weird that the layout doesn’t change. The subject area is designed to match the height of the toolbar, so I assumed the SVG subject height would shrink to allow for the height of the toolbar in the subject area. 

My concern was that this changes the height of the subject for live projects. If that’s not happening, then there’s no problem.

**Comment by eatyourgreens** [comment]
By the way, if you want to try out design ideas for new layouts, but don’t want to change PH-TESS etc., then adding a new layout to the layouts map would give you a layout that you can experiment with while also controlling which projects and workflows use it. 

That might be useful for testing big changes, rather than editing the default layout directly.

File: packages/lib-classifier/src/components/Classifier/components/MetaTools/MetaTools.js
Comment (mcbouslog) [inline packages/lib-classifier/src/components/Classifier/components/MetaTools/MetaTools.js:61]: When I run locally, signed in, and load https://local.zooniverse.org:3000/projects/brooke/i-fancy-cats/classify/workflow/693 or https://local.zooniverse.org:3000/projects/pmlogan/poets-and-lovers/classify/workflow/21362/subject-set/104809?env=production the Metadata, FavouritesButton, and CollectionsButton are unexpectedly disabled. Console logging the subject from MetaTools returns `null` and the component doesn't appear to rerender once the subject (and related isThereMetadata) is defined.

File: packages/lib-classifier/src/components/Classifier/components/MetaTools/MetaTools.stories.js
Comment (mcbouslog) [inline packages/lib-classifier/src/components/Classifier/components/MetaTools/MetaTools.stories.js:71]: I'm not sure what this shows, maybe a comment to help clarify what the red box represents and how the MetaTools component should look or change from the default? 

If the idea is to show what the classifier looks like on a small screen, I changed the layout to an iPhone12 in the storybook and the buttons looked good and as expected. I'm not sure if it's possible, but maybe changing the defaultLayout just for this story to an iPhone or mobile dimensions might be helpful?

**Review by mcbouslog — CHANGES_REQUESTED** [review]
Looks good! Big improvement for small screens. Looked strange to me at first on desktop, but only because I'm used to current grid.

Refactors look good 👍 . Only blocking comment is regarding MetaTools missing observer (I think), otherwise a few other comments for consideration, but not blocking.

**Comment by mcbouslog** [comment]
Unrelated to these changes, but while reviewing I noticed the Hide Previous Marks svg icon is 24px while the other MetaTools icons are 15px. The difference isn't noticeable (to me anyway) on desktop, but when stacked for a small screen the alignment is a little more noticeable. Not sure if want to include a fix for in these changes, we can convert this comment to an issue, or other option.

File: packages/lib-classifier/src/components/Classifier/components/MetaTools/MetaTools.js
Comment (goplayoutside3) [reply packages/lib-classifier/src/components/Classifier/components/MetaTools/MetaTools.js:64]: You're right! I think adding `observer` restored some of the expected functionality, but I also noticed that you can visit [i-fancy-cats on frontend.preview](https://frontend.preview.zooniverse.org/projects/brooke/i-fancy-cats?env=staging), sign-in to enable FavouritesButton and CollectionsButton, but then upon sign-out the `upp` from the store still contains user data, and the buttons are all still enabled. I think this is a bug of the UserProjectPreferences store so I'll do a quick search for an already existing Github Issue or open a new one.

File: packages/lib-classifier/src/components/Classifier/components/MetaTools/components/Metadata/components/MetadataButton/MetadataButton.spec.js
Comment (goplayoutside3) [reply packages/lib-classifier/src/components/Classifier/components/MetaTools/components/Metadata/components/MetadataButton/MetadataButton.spec.js]: Nice! I'll update this unit test accordingly

File: packages/lib-classifier/src/components/Classifier/components/MetaTools/MetaTools.stories.js
Comment (goplayoutside3) [reply packages/lib-classifier/src/components/Classifier/components/MetaTools/MetaTools.stories.js]: I think the border was left over from building the story, since it just makes the containing `div` easier to see. The idea of this story is to show what the Classifier's subject grid area looks like when its containing `div` is small. Remember that this case does not occur only on mobile screens. The default layout is 2-column, and both columns shrink in width as a browser window decreases in width.

I removed the border.

**Comment by goplayoutside3** [comment]
>  ...icon is 24px while the other MetaTools icons are 15px. The difference isn't noticeable (to me anyway) on desktop, but when stacked for a small screen the alignment is a little more noticeable. 

I agree the difference is annoying when all meta buttons are stacked on top of each other when its column is of smaller width. I plan to follow-up with Sean to get his approval on this PR as well, and my review with him will include minor adjustments of icon size, margins, etc

**Review by mcbouslog — APPROVED** [review]
Changes LGTM 👍

**Comment by goplayoutside3** [comment]
Met with @seanmiller26 to review MetaTools design. A few notes on what will follow in a separate PR:
- Align all meta buttons if stacked on top of each other
- Work toward minor refactor of meta tools button responsive design (will open an Issue with design reference)
- Make the MultiFrameViewer's Filmstrip sticky like the image toolbar

**Pull request merged**
